noir/noir-repo/aztec_macros/src/lib.rs

```diff
@@ -506,6 +517,128 @@ fn transform_module(
     Ok(has_transformed_module)
 }
 
+fn check_trait_method_implemented(trait_impl: &NoirTraitImpl, method_name: &str) -> bool {
+    trait_impl.items.iter().any(|item| match item {
+        TraitImplItem::Function(func) => func.def.name.0.contents == method_name,
+        _ => false,
+    })
+}
+
+fn generate_note_interface_impl(module: &mut SortedModule) -> Result<(), AztecMacroError> {
```

Comment:
Can we comment above this method how we transform things?

Reply:
Perhaps we should just make it such that `NoteInterface` should implemented using a macro rather than checking whether methods exist in this way. 
```
ValueNote {
     address: Field,
     value: Field,
}
```
Then this code:
```
impl NoteInterface for ValueNote {
    // does impl here
}
```
will get transformed to this:
```
#[aztec(note_interface)]
note_impl(ValueNote, NoteInterface, address, note);
```
In the package when we see `#[aztec(note_interface)]` we should then implement the macro with all the methods.

Reply:
Or we just need to make a `derive` macro.

Reply:
Yeah, that's probably the final version of this, but I wanted to check with you the "access the whole AST in the macro" problem before continuing.